capi/geos_ts_c.cpp

```diff
@@ -2977,9 +2979,15 @@ extern "C" {
 
         return execute(extHandle, [&]() {
             PrecisionModel newpm;
-            if(gridSize != 0) {
-                // Use negative scale to indicate you actually want a gridSize
-                double scale = -1.0 * std::abs(gridSize);
+            if (gridSize != 0) {
+                //-- check for an integral scale
+                double scale = 1.0 / gridSize;
+                //-- add a small "bump" to ensure flooring works
+                double scaleInt = std::floor(scale + 2 * GRIDSIZE_INTEGER_TOLERANCE);
```

Comment:
I'm probably missing something obvious, but why not `std::round` here?

Reply:
yes, much better idea.